qualang_tools/video_mode/videomode.py

```diff
@@ -0,0 +1,452 @@
+# VideoMode script enabling the update of parameters in a QUA program through user input while the program is running.
+# Authors: Arthur Strauss, Théo Laudat
+# Date: 19/12/2023
+
+import threading
+from qm.qua import *
+from qm import QuantumMachine, QmJob, Program
+import time
+import signal
+from typing import Optional, List, Dict, Union
+from dataclasses import dataclass
+import numpy as np
+
+
+@dataclass
+class ParameterTable:
+    """
+    Data class enabling the mapping of parameters to be updated to their corresponding "to-be-declared" QUA variables.
+    The type of the QUA variable to be adjusted is automatically inferred from the type of the initial_parameter_value.
+    Each parameter in the dictionary should be given a name that the user can then easily access through the table with
+    table[parameter_name]. Calling this will return the QUA variable built within the QUA program corresponding to the parameter name
+    and its associated Python initial value.
+    Args:
+        parameters_dict: Dictionary of the form { "parameter_name": initial_parameter_value }.
+        the QUA program.
+    """
+
+    parameters_dict: Dict[str, Union[float, int, bool, List, np.ndarray]]
+
+    def __post_init__(self):
+        self.table = {}
+        for index, (parameter_name, parameter_value) in enumerate(
+            self.parameters_dict.items()
+        ):
+            self.table[parameter_name] = {"index": index, "value": parameter_value}
+            if isinstance(parameter_value, float):
+                if float(parameter_value).is_integer() and parameter_value > 8:
+                    self.table[parameter_name][
+                        "declare_expression"
+                    ] = f"declare(int, value=int({parameter_value}))"
```

Comment:
Yes I understand your point, however the idea is to ask the minimum from the user so that it will be easy for him/her to implement and I had in mind that `var=0.0 --> fixed` and `var=0 --> int`.
We can also add the type of the QUA variable as a key of `param_dict`, what do you think? @arthurostrauss ?

Reply:
I agree with Yonatan on this one. I think that declaring the type is not a large overhead

Reply:
So you would like the input param_dict to be of the form:
`{param_name:
{"value": initial_value,
"type": type(initial_value)}
}`

Is that correct?  
The overhead to declare the type of each variable attached to each parameter in the input dict is in my opinion an unnecessary redundancy, as if the user has lots of parameters, all initialised to Python values, he will in most cases want to do type(parameter_value) anyway. The error appears later in the code if he tries to update an int with a float variable for example. 

I think the extra attention to let the user know that he should define carefully the initial_value is enough to avoid the explicit type definition of the variable type (especially because we have a limited set of such types).

Reply:
I think that if we make the user specify the type, then it must be a "QUA" type where the only possibilities are `int`, `bool` or `fixed`, no matter the initial values. It is true that now, if you initialize a variable with 0, then it will be declared as an `int`, which may not be what the user wants...
I the majority thinks it is better (more robust and less confusing for the user) to specify the type, then I am also for it

Reply:
I suggest forcing the type as being something optional. That means that we maintain compatibility with the current inferred type as done right now, and on top of it, if the user specifies the type we do an additional check that the value matches the provided input type to declare the QUA variable

Reply:
I'm against automatic inferring of the variable type, as I believe it will cause errors, and the error message will not be indicative enough: A user will get an error that duration cannot accept a fixed point parameter, but he will have no way of knowing how to fix it. Example of error message:
`2023-12-24 13:28:20,824 - qm - ERROR    - Expected type int but got fixed (@ File "D:\GitHub\qm-cs-yoav\IQM\IQ_blobs.py", line 361: play("pi_half", "q1_xy", duration=I) )`

Moreover, users are used to declaring the variable type.

But with that being said, I'm leaving this up to CS to decide, so if both @TheoLaudatQM and @arthurostrauss want to automatically infer it, go ahead.

Reply:
I modified the code to take into account the forced type specification. I left available the possibility to infer it automatically but I moved it away from the documentation for now. All examples and tests are updated accordingly